app/assets/javascripts/discourse/app/components/user-menu/notifications-list.js

```diff
@@ -6,19 +6,25 @@ import {
   mergeSortedLists,
   postRNWebviewMessage,
 } from "discourse/lib/utilities";
-import showModal from "discourse/lib/show-modal";
 import { inject as service } from "@ember/service";
 import UserMenuNotificationItem from "discourse/lib/user-menu/notification-item";
 import Notification from "discourse/models/notification";
 import UserMenuReviewable from "discourse/models/user-menu-reviewable";
 import UserMenuReviewableItem from "discourse/lib/user-menu/reviewable-item";
+import DismissNotificationConfirmationModal from "discourse/components/modal/dismiss-notification-confirmation";
+
+let alwaysDisplayDismissWarning;
+export function displayDismissNotificationConfirmation() {
+  alwaysDisplayDismissWarning = true;
+}
```

Comment:
If I'm reading this correctly then, after this new API is called by a plugin, confirmations will be force-enabled globally? Even for tabs which are unrelated to the plugin? Is that the intention?

It feels very strange that plugins should be able to affect other tabs. I would've expected the assign plugin to set the `renderDismissConfirmation()` value in its own tab component, just the same as in core 🤔

Reply:
Dropped the additions to the plugin api in favor of a getter than can be overridden in a notification item component